src/ws/Socket.ts

```diff
@@ -464,6 +465,7 @@ export class Socket extends EventEmitter {
             // tslint:disable-next-line no-floating-promises
             this.call(authMethod, this._authpacket, { force: true })
                 .then(result => this.emit('authresult', result))
+                .then(() => this.optOutEvents(this._optOutEventsArgs))
```

Comment:
You should only call this if the user opted out of events previously. Also you may want to send with `force: true` to avoid spooling it multiple times if the socket disconnects before you get a reply back.

Reply:
Sure, I could store that information if the user has opted out of any events, but ideally this won't do anything as `this._optOutEventsArgs` would be empty. And yeah I can pass in `force: true`, that's my bad, first time touching any code in this library and never used it from frontend too :)

Reply:
No worries, this is a rather... involved, bit of code.

Ideally opt out would do nothing if empty, true, but no need to make the extra calls  :)